R/merge_sce_list.R

```diff
@@ -118,6 +129,41 @@ merge_sce_list <- function(sce_list = list(),
   # Create the merged SCE from the processed list ------------------
   merged_sce <- do.call(cbind, sce_list)
 
+  # combine library metadata into a single data frame
+  metadata_list <- metadata(merged_sce)
+
+  # create a single list of library metadata
+  library_metadata <- metadata_list[names(metadata_list) == "library_metadata"]
+
+  # get a single list of all library ids
+  library_ids <- metadata(merged_sce)[names(metadata_list) == "library_id"] |>
+    unlist() |>
+    unique()
+
+  # get a single list of all sample ids
+  sample_ids <- metadata(merged_sce)[names(metadata_list) == "sample_id"] |>
+    unlist() |>
+    unique()
```

Comment:
```suggestion
  # get a vector of all library ids
  library_ids <- sce_list |> 
    purrr::map("library_id") |>
    unlist() |>
    unique()

  # get a vector of all sample ids
  sample_ids <- sce_list |> 
    purrr::map("sample_id") |>
    unlist() |>
    unique() 
```

Reply:
I just want to confirm that what you mean is below. We want to pull from the `metadata_list` and not the `sce_list`? 
```
  library_ids <- metadata_list |> 
    purrr::map("library_id") |>
    unlist() |>
    unique()
```

Reply:
As I was writing what I really wanted, I just realized we can do most of this in a single step with `purrr::transpose()` to turn the list of lists "inside out". At least I think so.

```
# get a list of just metadata and invert it so each library is an element within the metadata 
# components from `prepare_sce_for_merge()`
metadata_list <- sce_list |>
  purrr::map(metadata) |>
  purrr::transpose()

# flatten and reduce the library ids and sample ids
metadata_list$library_id <- metadata_list$library_id |>
  unlist() |>
  unique()
metadata_list$sample_id<- metadata_list$sample_id |>
  unlist() |>
  unique()

# row bind the sample metadata (I didn't add the if statement here)
metadata_list$sample_metadata <- metadata_list$sample_metadata |>
  dplyr::bind_rows() |>
  unique()
```
Does that make sense?

Then you can remove lines 150 - 155 as they are already combined into a list

Reply:
Yes this makes sense! Thank you for this, I think this really cleans it up! I implemented this in https://github.com/AlexsLemonade/scpcaTools/pull/213/commits/6024762765bfc2c96a28c189f2bee7b2a38d132b.